docs/src/snap/howto/cis-hardening.md

```diff
@@ -282,6 +347,3230 @@ the sets, including the dqlite specific checks in the output.
+/var/snap/k8s/common/args/kubelet on each worker node and set the following
+argument.
+
+`--read-only-port=0`
+
+Restart the kubelet service:
+
+`snap restart k8s.kubelet`
+
+
+**Audit (as root):**
+
+```
+/bin/cat /var/snap/k8s/common/args/kubelet
+```
+
+**Expected output:**
+
+```
+--read-only-port=0
+```
+
+##### Control 4.2.5
+
+**Description:**
+
+Ensure that the --streaming-connection-idle-timeout argument
+is not set to 0 (Manual)
+
+
+**Remediation:**
+
+Edit the kubelet configuration file
+/var/snap/k8s/common/args/kubelet on each worker node and set the following
+argument.
+
+`--streaming-connection-idle-timeout=5m`
+
+Restart the kubelet service:
+
+`snap restart k8s.kubelet`
+
+
+**Audit (as root):**
+
+```
+/bin/cat /var/snap/k8s/common/args/kubelet
+```
+
+**Expected output:**
+
+```
+--streaming-connection-idle-timeout is not set, or set to a
+value greater or equal than 5m
+```
+
+##### Control 4.2.6
+
+**Description:**
+
+Ensure that the --protect-kernel-defaults argument is set to
+true (Automated)
+
+
+**Remediation:**
+
+Edit the kubelet configuration file
+/var/snap/k8s/common/args/kubelet on each worker node and set the following
+argument:
+
+`--protect-kernel-defaults=true`
+
+Restart the kubelet service:
+
+`snap restart k8s.kubelet`
+
+
+**Audit (as root):**
+
+```
+/bin/cat /var/snap/k8s/common/args/kubelet
+```
+
+**Expected output:**
+
+```
+--protect-kernel-defaults=true
+```
+
+##### Control 4.2.7
+
+**Description:**
+
+Ensure that the --make-iptables-util-chains argument is set
+to true (Automated)
+
+
+**Remediation:**
+
+Edit the kubelet configuration file
+/var/snap/k8s/common/args/kubelet on each worker node and
+set the following argument:
+
+`--make-iptables-util-chains=true`
+
+Restart the kubelet service.
+
+For example: `snap restart k8s.kubelet`
+
+
+**Audit (as root):**
+
+```
+/bin/cat /var/snap/k8s/common/args/kubelet
+```
+
+**Expected output:**
+
+```
+--make-iptables-util-chains is not set
+```
+
+##### Control 4.2.8
+
+**Description:**
+
+Ensure that the --hostname-override argument is not set
+(Manual)
+
+
+**Remediation:**
+
+Edit the kubelet configuration file /var/snap/k8s/common/args/kubelet
+on each worker node and remove the --hostname-override
+argument.
+
+Restart the kubelet service.
+
+`snap restart k8s.kubelet`
+
+
+**Audit (as root):**
+
+```
+/bin/cat /var/snap/k8s/common/args/kubelet
+```
+
+**Expected output:**
+
+```
+--hostname-override is set to false
+```
+
+##### Control 4.2.9
+
+**Description:**
+
+Ensure that the --event-qps argument is set to a level which
+ensures appropriate event capture (Manual)
+
+
+**Remediation:**
+
+Edit the kubelet configuration file /var/snap/k8s/common/args/kubelet on each
+worker node and
+set the --event-qps parameter as appropriate.
+
+Restart the kubelet service.
+
+`snap restart k8s.kubelet`
+
+
+**Audit (as root):**
+
+```
+/bin/cat /var/snap/k8s/common/args/kubelet
+```
+
+**Expected output:**
+
+```
+--event-qps is not set
```

Comment:
expected output does not match remediation

Reply:
See https://github.com/canonical/k8s-snap/pull/802/files#diff-da8b48161c214d6854a37a2ef8a664491413a6b43793ba6e9e6319a49da719ecR2951